pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -684,23 +684,29 @@ func (c *Cloud) updateInstanceSecurityGroupsForNLBTraffic(actualGroups []*ec2.Se
 				}
 
 				// MTU discovery
-				permission = &ec2.IpPermission{
-					IpProtocol: aws.String("icmp"),
-					FromPort:   aws.Int64(3),
-					ToPort:     aws.Int64(4),
-				}
-				ranges = []*ec2.IpRange{}
-				for _, cidr := range clientCidrs {
-					ranges = append(ranges, &ec2.IpRange{
-						CidrIp:      aws.String(cidr),
-						Description: aws.String(mtuRuleAnnotation),
-					})
-				}
-				permission.IpRanges = ranges
-				if add {
-					adds = append(adds, permission)
-				} else {
-					removes = append(removes, permission)
+				if !icmpPermAdded {
+					// avoid adding the permission more than once
+					mtuRuleAnnotation := fmt.Sprintf("%s=%s", NLBMtuDiscoveryRuleDescription, lbName)
+					permission = &ec2.IpPermission{
+						IpProtocol: aws.String("icmp"),
+						FromPort:   aws.Int64(3),
+						ToPort:     aws.Int64(4),
+					}
+					ranges = []*ec2.IpRange{}
+					for _, cidr := range clientCidrs {
+						ranges = append(ranges, &ec2.IpRange{
+							CidrIp:      aws.String(cidr),
+							Description: aws.String(mtuRuleAnnotation),
+						})
+					}
+					permission.IpRanges = ranges
+					if add {
+						adds = append(adds, permission)
+					} else {
+						removes = append(removes, permission)
```

Comment:
If we have 2 ports and we remove one, we don't want to remove the MTU rule, do we?

I suspect the MTU rule belongs outside the port changes loop.